azure/datalake/store/transfer.py

```diff
@@ -269,6 +269,14 @@ def submit(self, src, dst, length):
             tmpdir = None
 
         offsets = list(range(0, length, self._chunksize))
+
+        # in the case of empty files, ensure that the initial offset of 0 is properly added.
+        if len(offsets) == 0 and length == 0:
```

Comment:
This and the below guards could be simplified by having a top-level `if len(offsets) == 0` and then have an `if` inside that has guards of `not length` and `else`. E.g.
```python
if not len(offsets):
  if not length:
    offsets.append(0)
  else:
    raise ...
```